### DISCUSSION TITLE
Is there any way of passing the expected format to the DatePicker but let the DatePicker empty?

### DISCUSSION THREAD

**Discussion by semoal** [adobe/react-spectrum#3531] [Q&A]
Hi!

We're using at our company the DatePicker, Calendar, and many more hooks of react-aria, but we're getting some limitations about how to handle this scenario.

1. We have a form that the user will submit some transactions that must contains date and time
2. The user should see the DatePicker empty.
3. On change, it should pick the date and time but without displaying the time picker.

We're using `granularity=day` and initialzing the default value with `now(getLocaltimezone())` but the we don't want to show this date, we want to "force" somehow the user to add a date and keeping it blank when submits.

Shouldn't be an option to enable which kind of date we want on the onChange callback?

**Comment by devongovett** [comment]
You always get back the same type of object as you put in. You can easily convert between the types in your `onChange` handler if you want.

Not sure I fully understood your exact requirements, but this example accepts a `ZonedDateTime` in `onChange`, and converts it to a `CalendarDate`, which excludes the time. See https://react-spectrum.adobe.com/internationalized/date/ZonedDateTime.html#to-a-date-or-time-only

```js
import {toCalendarDate} from '@internationalized/date';

let onChange = (dateTime) => {
  let dateOnly = toCalendarDate(dateTime);
  // ...
};
```
